src/Http/Http.Extensions/gen/StaticRouteHandlerModel/EndpointParameter.cs

```diff
@@ -389,6 +451,74 @@ private static string GetEscapedParameterName(AttributeData attribute, string pa
         }
     }
 
+    private static bool TryGetAsParametersConstructor(INamedTypeSymbol type, out bool? isParameterlessConstructor, [NotNullWhen(true)] out IEnumerable<ConstructorParameter>? matchedParameters)
+    {
+        isParameterlessConstructor = null;
+        matchedParameters = null;
+        if (type.IsAbstract)
+        {
+            return false;
+        }
+
+        var constructors = type.Constructors.Where(constructor => constructor.DeclaredAccessibility == Accessibility.Public && !constructor.IsStatic);
+        var numOfConstructors = constructors.Count();
+        // When leveraging parameterless constructors, we want to ensure we only emit for writable
+        // properties. We do not have this constraint if we are leveraging a parameterized constructor.
+        var properties = type.GetMembers().OfType<IPropertySymbol>().Where(property => property.DeclaredAccessibility == Accessibility.Public);
+        var writableProperties = properties.Where(property => !property.IsReadOnly);
+
+        if (numOfConstructors == 1)
+        {
+            var targetConstructor = constructors.SingleOrDefault();
+            var lookupTable = new Dictionary<ParameterLookupKey, IPropertySymbol>();
+            foreach (var property in properties)
+            {
+                lookupTable.Add(new ParameterLookupKey(property.Name, property.Type), property);
+            }
+
+            var parameters = targetConstructor.GetParameters();
+            var propertiesWithParameterInfo = new List<ConstructorParameter>();
+
+            if (parameters.Length == 0)
+            {
+                isParameterlessConstructor = true;
+                matchedParameters = writableProperties.Select(property => new ConstructorParameter(property, null));
+                return true;
+            }
+
+            foreach (var parameter in parameters)
+            {
+                var key = new ParameterLookupKey(parameter.Name!, parameter.Type);
+                if (lookupTable.TryGetValue(key, out var property))
+                {
+                    propertiesWithParameterInfo.Add(new ConstructorParameter(property, parameter));
+                }
+                else
+                {
+                    return false;
+                }
+            }
+
+            isParameterlessConstructor = false;
+            matchedParameters = propertiesWithParameterInfo;
+            return true;
+        }
+
+        if (type.IsValueType)
+        {
+            isParameterlessConstructor = true;
+            matchedParameters = writableProperties.Select(property => new ConstructorParameter(property, null));
+            return true;
+        }
+
+        if (numOfConstructors > 1)
```

Comment:
Is this just for documentation purposes?  It looks like it behaves the same as the default case (0 parameters?)?

Reply:
Yes, in the latest commit (https://github.com/dotnet/aspnetcore/pull/47914/commits/26d174a3b396be0ef6fd57c2b0fc93ae6f582a46), I've added the logging to the RDG for this scenario that typically happens in the RDF.